### DISCUSSION TITLE
Includes tqdm bar

### DISCUSSION THREAD

**Discussion by mreismendes** [winedarksea/AutoTS#26] [Ideas]
It would be interesting to see an estimation of the number of models to be trained and the concluded percentage.

Using TQDM would be great.

PS: I'm not a software engineer. I'm just a data scientist that sometimes goes deep into under the hood issues.

Thank you!

**Comment by winedarksea** [comment]
I could do a bar to show number of _generations_ but not _number of models_. The reason for that is models are generated dynamically for each generation. Actual number of models per generation is approximately the same - say N models (it's about 50 max if I remember) are created, these can include duplicates of previous models, these duplicates are then removed which leaves you usually with N minus some random amount in each generation.
I feel I kind-of do this already with the print messages (if verbose > 0). I am not sure how much tqdm would actually add to the experience - sure it would be prettier, but it's also another dependency.

**Reply by mreismendes** [reply]
I know there is an option to built more than one level of "TQDM" as far as you know the n models previously. But as you said I think this enhancement surely is not on the top of the bucket list, it's more related to aesthetical issues.
